src/amo/components/AddonReview.js

```diff
@@ -0,0 +1,135 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import { asyncConnect } from 'redux-connect';
+
+import { submitReview } from 'amo/api';
+import { setReview } from 'amo/actions/reviews';
+import { callApi } from 'core/api';
+import translate from 'core/i18n/translate';
+
+import 'amo/css/AddonReview.scss';
+
+
+export class AddonReviewBase extends React.Component {
+  static propTypes = {
+    apiState: PropTypes.object,
+    i18n: PropTypes.object.isRequired,
+    review: PropTypes.object.isRequired,
+    updateReviewText: PropTypes.func.isRequired,
+  }
+
+  static contextTypes = {
+    router: PropTypes.object.isRequired,
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    const body = this.reviewTextarea.value;
+    if (!body) {
+      // TODO: dispatch an error action when we have support for that.
+      throw new Error('The review text cannot be empty');
+    }
+    const params = {
+      body,
+      addonId: this.props.review.addonSlug,
+      reviewId: this.props.review.id,
+      apiState: this.props.apiState,
+      router: this.context.router,
+    };
+    // TODO: render a progress indicator in the UI.
+    this.props.updateReviewText(params);
+  }
+
+  render() {
+    const { i18n, review } = this.props;
+    if (!(review && review.id && review.addonSlug)) {
+      throw new Error(`Unexpected review property: ${JSON.stringify(review)}`);
+    }
+
+    // TODO: I guess we should load the existing review text so it
+    // can be edited? That flow needs more thought.
+    return (
+      <div className="AddonReview">
+        <h2>{i18n.gettext('Write a review')}</h2>
+        <p>
+          {i18n.gettext(
+            'Tell the world why you think this extension is fantastic!'
+          )}
+        </p>
+        <form onSubmit={this.onSubmit}>
+          <textarea
+            ref={(ref) => { this.reviewTextarea = ref; }}
+            name="review"
+            placeholder={i18n.gettext(
+              'Tell us what you love about this extension. ' +
+              'Be specific and concise.')} />
+          <div className="AddonReview-button-row">
+            {/* TODO: hook up the back button when we have link support */}
+            <button className="AddonReview-button AddonReview-back-button">
+              &#12296; {i18n.gettext('Back')}
```

Comment:
Do we have any examples of inserting content like `&#12296;` with sass? This is turning into an escaping nightmare.